lib/core.ps1

```diff
@@ -577,90 +577,114 @@ function shim($path, $global, $name, $arg) {
-            "# $resolved_path
-if (!(Test-Path Variable:PSScriptRoot)) { `$PSScriptRoot = Split-Path `$MyInvocation.MyCommand.Path -Parent }
-`$path = join-path `"`$psscriptroot`" `"$relative_path`"
-if(`$myinvocation.expectingInput) { `$input | & `$path $arg @args } else { & `$path $arg @args }
-exit `$lastexitcode"
+            @(
+                "# $resolved_path",
+                "if (!(Test-Path Variable:PSScriptRoot)) { `$PSScriptRoot = Split-Path `$MyInvocation.MyCommand.Path -Parent }",
+                "`$path = Join-Path `"`$PSScriptRoot`" `"$relative_path`"",
+                "if (`$MyInvocation.ExpectingInput) { `$input | & `$path $arg @args } else { & `$path $arg @args }",
+                "exit `$LASTEXITCODE"
+            )
         }
-        $ps1text | Out-File "$shim.ps1" -Encoding ASCII
+        $ps1text -join "`r`n" | Out-File "$shim.ps1" -Encoding ASCII
 
         # make ps1 accessible from cmd.exe
         warn_on_overwrite "$shim.cmd" $path
-        "@rem $resolved_path
-@echo off
-setlocal enabledelayedexpansion
-set args=%*
-:: replace problem characters in arguments
-set args=%args:`"='%
-set args=%args:(=``(%
-set args=%args:)=``)%
-set invalid=`"='
-if !args! == !invalid! ( set args= )
-where /q pwsh.exe
-if %errorlevel% equ 0 (
-    pwsh -noprofile -ex unrestricted -command `"& '$resolved_path' $arg %args%;exit `$lastexitcode`"
-) else (
-    powershell -noprofile -ex unrestricted -command `"& '$resolved_path' $arg %args%;exit `$lastexitcode`"
-)" | Out-File "$shim.cmd" -Encoding ASCII
+        @(
+            "@rem $resolved_path",
+            '@echo off',
+            'setlocal enabledelayedexpansion',
+            'set args=%*',
+            ':: replace problem characters in arguments',
+            "set args=%args:`"='%",
+            "set args=%args:(=``(%",
+            "set args=%args:)=``)%",
+            "set invalid=`"='",
+            'if !args! == !invalid! ( set args= )',
+            'where /q pwsh.exe',
+            'if %errorlevel% equ 0 (',
+            "    pwsh -noprofile -ex unrestricted -command `"& '$resolved_path' $arg %args%;exit `$lastexitcode`"",
+            ') else (',
+            "    powershell -noprofile -ex unrestricted -command `"& '$resolved_path' $arg %args%;exit `$lastexitcode`"",
+            ')'
+        ) -join "`r`n" | Out-File "$shim.cmd" -Encoding ASCII
 
         warn_on_overwrite $shim $path
-        "#!/bin/sh
-# $resolved_path
-if command -v pwsh.exe &> /dev/null; then
-    pwsh.exe -noprofile -ex unrestricted -command `"& '$resolved_path' $arg $@;exit \`$lastexitcode`"
-else
-    powershell.exe -noprofile -ex unrestricted -command `"& '$resolved_path' $arg $@;exit \`$lastexitcode`"
-fi" | Out-File $shim -Encoding ASCII
+        @(
+            '#!/bin/sh',
+            "# $resolved_path",
+            'if command -v pwsh.exe &> /dev/null; then',
+            "    pwsh.exe -noprofile -ex unrestricted -command `"& '$resolved_path' $arg $@;exit \`$lastexitcode`"",
```

Comment:
```suggestion
            "    pwsh -noprofile -ex unrestricted -command `"& '$resolved_path' $arg $@;exit \`$lastexitcode`"",
```

Reply:
Are there some benefits to trim `.exe`? I'm not sure the difference of using or not the extension of exe file.